crates/hc_sandbox/src/calls.rs

```diff
@@ -570,35 +580,135 @@ async fn call_inner(client: &mut AdminWebsocket, call: AdminRequestCli) -> anyho
                 // TODO FIXME
                 #[allow(deprecated)]
                 let exp = DateTime::<Utc>::from_utc(NaiveDateTime::from_timestamp(s, n), Utc);
-                let now = Utc::now();
-
-                writeln!(out, "signed at {}", dt)?;
-                writeln!(
-                    out,
-                    "expires at {} in {}mins",
-                    exp,
-                    (exp - now).num_minutes()
-                )?;
-                writeln!(out, "space: {:?}", info.space)?;
-                writeln!(out, "agent: {:?}", info.agent)?;
-                writeln!(out, "URLs: {:?}", info.url)?;
-                msg!("{}\n", out);
+
+                out.push(AgentResponse {
+                    agent_pub_key: this_agent.map(|a| a.0.clone().into()),
+                    k2_agent: this_agent.map(|a| a.1.clone()),
+                    dna_hash: this_dna.0.clone().into(),
+                    k2_space: this_dna.1.clone(),
+                    signed_at: dt,
+                    expires_at: exp,
+                    url: info.url.clone(),
+                });
             }
+            println!("{}", serde_json::to_value(&out)?);
         }
         AdminRequestCli::AgentMetaInfo(args) => {
             let info = client.agent_meta_info(args.url, args.dna).await?;
             let string_key_info = info
                 .into_iter()
-                .map(|(k, v)| (DnaHashB64::from(k).to_string(), v))
+                .map(|(k, v)| (k.to_string(), v))
                 .collect::<BTreeMap<String, BTreeMap<String, AgentMetaInfo>>>();
-
-            let info_json = serde_json::to_string_pretty(&string_key_info)?;
-            println!("{}", info_json);
+            println!("{}", serde_json::to_value(&string_key_info)?);
         }
     }
     Ok(())
 }
 
+/// Convert an [AppInfo] to JSON with extra base64 conversion of `agent_pub_key` and `cell_id` fields.
+fn convert_app_info(app_info: AppInfo) -> anyhow::Result<serde_json::Value> {
+    let value = serde_json::to_value(&app_info)?;
+    let serde_json::Value::Object(mut app_info_map) = value else {
+        bail!("Invalid appInfo conversion result");
+    };
+    // Convert `agent_pub_key` field
+    if let Some(old_value) = app_info_map.get("agent_pub_key") {
+        let new_value = transform_agent_pub_key(old_value.clone())?;
+        app_info_map.insert("agent_pub_key".to_string(), new_value);
+    }
+    // Convert `cell_info` field
+    if let Some(old_value) = app_info_map.get("cell_info") {
+        let new_value = transform_cell_info_map(old_value.clone())?;
+        app_info_map.insert("cell_info".to_string(), new_value);
+    }
+    //
+    Ok(serde_json::Value::Object(app_info_map))
+}
+
+/// Convert the inner `cell_id` fields of the JSON value of a `IndexMap<RoleName, Vec<CellInfo>>`.
+fn transform_cell_info_map(value: serde_json::Value) -> anyhow::Result<serde_json::Value> {
```

Comment:
```suggestion
fn cell_info_map_to_json(value: serde_json::Value) -> anyhow::Result<serde_json::Value> {
```

Reply:
Here the input is already JSON, so an explicit function name would rather be `convert_cell_id_to_base64_within_cell_info_map()`